iter/filter_test.go

```diff
@@ -53,7 +53,7 @@ func ExampleFilterMap() {
 }
 
 func ExampleExclude() {
-	filtered := iter.Exclude[int](iter.Lift([]int{0, 1, 0, 2}), filters.IsZero[int])
+	filtered := iter.Lift([]int{0, 1, 0, 2}).Exclude(filters.IsZero[int])
 	fmt.Println(filtered.Next())
 	fmt.Println(filtered.Next())
 	fmt.Println(filtered.Next())
```

Comment:
Keep this as it was, but create a second example `ExampleIter_method` that demonstrates what you're showing here.